tests/Unit/Models/FamilyModelTest.php

```diff
@@ -70,4 +72,67 @@ public function itHasAnAttributeThatReturnsNearestDueDateOrNull()
         // Test we've not got an expecting date
         $this->assertEquals($pregnancy->dob, $pregnant_family->expecting);
     }
+
+    /** @test */
+    public function itCanGenreateAndSetAnRvidCorrectly()
+    {
+        // Set up some families and centres.
+        $centre1 = factory(Centre::class)->create();
+        $centre2 = factory(Centre::class)->create();
+
+        $family1 = factory(Family::class)->create();
+        $family2 = factory(Family::class)->create();
+        $family3 = factory(Family::class)->create();
+
+        // Generate the RVIDs
+        $family1->generateRVID($centre1);
+        $family1->save();
+
+        $family2->generateRVID($centre1);
+        $family2->save();
+
+        $family3->generateRVID($centre2);
+        $family3->save();
+
+        // Check the fields have been set
+
+        // In Family1, sequence should be 1
+        $this->seeInDatabase('families', [
+            'id' => $family1->id,
+            'initial_centre_id' => $centre1->id,
+            'centre_sequence' => 1,
+        ]);
+
+        // In Family2, sequence should be 2
+        $this->seeInDatabase('families', [
+            'id' => $family2->id,
+            'initial_centre_id' => $centre1->id,
+            'centre_sequence' => 2,
+        ]);
+
+        // In Family3, sequence should be 1
+        $this->seeInDatabase('families', [
+            'id' => $family3->id,
+            'initial_centre_id' => $centre2->id,
+            'centre_sequence' => 1,
+        ]);
+    }
+
```

Comment:
Nice - though Technically you should test all the possible outcomes - including when you don't have a valid centre and when you find family with same centre - but that family has no centre_sequence. And when you give it a family that already has an RVID.

I'd be happy with those cases just added as todo.